packages/sitecore-jss-react/src/components/Date.tsx

```diff
@@ -21,39 +23,48 @@ export interface DateFieldProps {
    */
   editable?: boolean;
   render?: (date: Date | null) => React.ReactNode;
+  /**
+   * -- Edit Mode Metadata --
+   *
+   * Custom element to render in Pages in Metadata edit mode if field value is empty
+   */
+  emptyValueEditingPlaceholder?: React.ComponentClass | React.FC;
 }
 
-export const DateField: React.FC<DateFieldProps> = withFieldMetadata(
-  ({ field, tag, editable = true, render, ...otherProps }) => {
-    if (!field || (!field.editable && !field.value)) {
-      return null;
-    }
+export const DateField: React.FC<DateFieldProps> = withFieldMetadata<DateFieldProps>(
+  withEmptyValueEditingPlaceholder<DateFieldProps>(
```

Comment:
instead of creating another HOC, can the empty stub logic be put into withFieldMetadata?

Reply:
hm both HOCs work for metadata mode, but are doing two very different things...  I feel it is better to leave those separate?

Reply:
I look at it this way: both HOCs are required for editing and both need to be active in metadata mode. Having them separate could lead more error prone setuo and increased maintenance complexity. And I'd rather avoid stacking HOCs if we need to add yet another metadata functinality for fields - JSS developers would need to remember to apply all of them too for their custom fields, and it would be a bad experience.
Rendering empty values for fields in editing is still part of managing fields in metadata - in that sense it fits into withFieldMetadata HOC

Reply:
I'd lean towards separate here ... In addition to Yavor's reasoning, these are composite in nature (one wraps the other in terms of React/DOM tree) so nesting them makes sense.

Reply:
@ambrauer should we consider introducing a composite HOC that is basically "all you need for wrapping fields in editing". I do think we need something like that to simplify custom implementations

Reply:
@art-alexeyenko I think having them separate actually gives more flexibility (e.g. for custom field types). The field metadata is quite generic but I can imagine you may want something custom for the empty value placeholder (e.g. "isEmpty" logic is different).